From 0177bb9b07975f31c38cebe7a9adb6dcbf9b434b Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Mon, 28 Sep 2020 07:58:24 -0400
Subject: [PATCH 1/2] Revert "Add double tap to sleep gesture"

This reverts commit 93df760b59de6bc93948726a3726242b2126a9ca.

  weird, but why without menu option?..
---
 AndroidManifest-common.xml                    |  2 +-
 src/com/android/launcher3/Workspace.java      | 20 -------------------
 .../touch/WorkspaceTouchListener.java         |  1 -
 3 files changed, 1 insertion(+), 22 deletions(-)

diff --git a/AndroidManifest-common.xml b/AndroidManifest-common.xml
index 01ad2a2e8..f08a79bc2 100644
--- a/AndroidManifest-common.xml
+++ b/AndroidManifest-common.xml
@@ -52,7 +52,7 @@
     <uses-permission android:name="android.permission.EXPAND_STATUS_BAR" />
     <uses-permission android:name="android.permission.MEDIA_CONTENT_CONTROL" />
     <uses-permission android:name="android.permission.USE_BIOMETRIC" />
-    <uses-permission android:name="android.permission.DEVICE_POWER" />
+
 
     <!--
     Permissions required for read/write access to the workspace data. These permission name
diff --git a/src/com/android/launcher3/Workspace.java b/src/com/android/launcher3/Workspace.java
index 96026c72e..b27be8720 100644
--- a/src/com/android/launcher3/Workspace.java
+++ b/src/com/android/launcher3/Workspace.java
@@ -54,15 +54,12 @@ import android.util.Log;
 import android.util.SparseArray;
 import android.view.LayoutInflater;
 import android.view.MotionEvent;
-import android.view.GestureDetector;
 import android.view.View;
 import android.view.ViewGroup;
 import android.view.ViewTreeObserver;
 import android.view.accessibility.AccessibilityNodeInfo;
 import android.widget.Toast;
 
-import com.android.internal.util.bootleggers.BootlegUtils;
-
 import com.android.launcher3.Launcher.LauncherOverlay;
 import com.android.launcher3.LauncherAppWidgetHost.ProviderChangedListener;
 import com.android.launcher3.LauncherStateManager.AnimationConfig;
@@ -252,8 +249,6 @@ public class Workspace extends PagedView<WorkspacePageIndicator>
     // Handles workspace state transitions
     private final WorkspaceStateTransitionAnimation mStateTransitionAnimation;
 
-    private GestureDetector mGestureListener;
-
     /**
      * Used to inflate the Workspace from XML.
      *
@@ -285,24 +280,9 @@ public class Workspace extends PagedView<WorkspacePageIndicator>
 
         // Disable multitouch across the workspace/all apps/customize tray
         setMotionEventSplittingEnabled(true);
-
-        context.enforceCallingOrSelfPermission(
-                    android.Manifest.permission.DEVICE_POWER, null);
-        mGestureListener =
-                new GestureDetector(context, new GestureDetector.SimpleOnGestureListener() {
-            @Override
-            public boolean onDoubleTap(MotionEvent event) {
-                BootlegUtils.switchScreenOff(context);
-                return true;
-            }
-        });
         setOnTouchListener(new WorkspaceTouchListener(mLauncher, this));
     }
 
-    public boolean checkDoubleTap(MotionEvent ev) {
-        return mGestureListener.onTouchEvent(ev);
-    }
-
     @Override
     public void setInsets(Rect insets) {
         DeviceProfile grid = mLauncher.getDeviceProfile();
diff --git a/src/com/android/launcher3/touch/WorkspaceTouchListener.java b/src/com/android/launcher3/touch/WorkspaceTouchListener.java
index 1d67bd669..310d598b2 100644
--- a/src/com/android/launcher3/touch/WorkspaceTouchListener.java
+++ b/src/com/android/launcher3/touch/WorkspaceTouchListener.java
@@ -81,7 +81,6 @@ public class WorkspaceTouchListener extends GestureDetector.SimpleOnGestureListe
     @Override
     public boolean onTouch(View view, MotionEvent ev) {
         mGestureDetector.onTouchEvent(ev);
-        mWorkspace.checkDoubleTap(ev);
 
         int action = ev.getActionMasked();
         if (action == ACTION_DOWN) {
-- 
2.17.1

